### ISSUE TITLE
`umount -c` a fuse-filesystem requires superuser privileges

### ISSUE THREAD

**Issue by fritz-k** [karelzak/util-linux#1192]
Hi,

when adding the `-c` flag to umount, I can't unmount a fuse filesystem as regular user:
```bash
$ umount /home/user/sshfs -c
umount: /home/user/sshfs:  must be superuser to unmount.
```
while
```bash
$ umount /home/user/sshfs
```
works fine.

Am I doing something wrong? Is this behaviour intended?

I'm trying to automount a fuse.sshfs filesystem with a systemd-user service. Systemd calls `umount -c` to unmount which leads to a failure stopping the mount-service. I'm running util-linux 2.36.

Cheers,
Simon

**Comment by karelzak** [comment]
We assume everywhere that mount paths are canonicalized, so -c is very special option which makes sense in case we trust to caller. It does not make sense to trust to non-root users.

Since v2.35 we drop suid permissions if any collision with internal umount(8) rules is detected then it continues as non-root user. 

It works as expected if the mount operation is redirected to some /sbin/mount.<type> helper, for example `mount -t fuse.sshfs`, the option`-t` forces mount(8) to drop suid, and then it calls /sbin/mount.fuse.sshfs as non-root user.

Unfortunately, for umount operation, there is not any helper and directly umount(2) syscall is requered, and it requires root permissions (unfortunately suid has been dropt due to `-c`).

For now the best solution is to somehow force systemd to not use -c for your service. The long therm solution is to improve libmount to ignore `-c` in this case.

**Comment by karelzak** [comment]
Implemented in the master branch (will be in v2.37). Not sure if we need to backport it to next maintanence release v2.36.2.

Thanks for your report!

**Comment by fritz-k** [comment]
Wow, thanks a lot for looking into my issue and pushing a fix so quickly!

Cheers,
Simon
